src/main/java/cms/gov/madie/measure/resources/MeasureTransferController.java

```diff
@@ -101,9 +104,27 @@ public ResponseEntity<Measure> createMeasure(
     // set ids for groups
     measure.getGroups().forEach(group -> group.setId(ObjectId.get().toString()));
     reorderGroupPopulations(measure.getGroups());
-    Measure savedMeasure = repository.save(measure);
-    measureSetService.createMeasureSet(
-        harpId, savedMeasure.getId(), savedMeasure.getMeasureSetId());
+
+    Measure savedMeasure = null;
+    List<Measure> existingMeasures =
+        measureTransferService.findByMeasureSetId(measure.getMeasureSetId());
+    if (!CollectionUtils.isEmpty(existingMeasures)) {
+      if (ModelType.QI_CORE.getValue().contains(measure.getModel())) {
+        throw new DuplicateMeasureException();
+      } else {
```

Comment:
This exception's condition can be executed earlier since we're going to always check for duplicates.

Reply:
@jkotanchik-SB Ok yeah makes sense, will do.

Reply:
@jkotanchik-SB moved it to the beginning of the method, thanks.

Reply:
@jkotanchik-SB It is good to add another condition before deleting and overwriting measure for qdm, but since in the very beginning of the method, an exception will be thrown for qi-core measures that have a list of the measures from db with the same measureSetId, therefore at this point, even if we don't add the condition of model being qdm, there won't be any qi-core measures being deleted or overwriten. Nevertheless, it's good to be specific.

Reply:
Ah, you right. Thanks for walking me through it!